qa/rpc-tests/quality_mempool.py

```diff
@@ -0,0 +1,424 @@
+
+        mark_logs("Node0 generating more blocks to achieve end of withdrawal epoch", self.nodes, DEBUG_MODE)
+        self.nodes[0].generate(EPOCH_LENGTH - 3)
+        self.sync_all()
+        assert_equal(self.nodes[0].getscinfo(scid_1)['items'][0]['balance'], creation_amount + fwt_amount) # Sc balance has matured
+        assert_equal(len(self.nodes[0].getscinfo(scid_1)['items'][0]['immature amounts']), 0)
+
+        epoch_block_hash_1, epoch_number_1 = get_epoch_data(scid_1, self.nodes[0], EPOCH_LENGTH)
+        mark_logs("epoch_number = {}, epoch_block_hash = {}".format(epoch_number_1, epoch_block_hash_1), self.nodes, DEBUG_MODE)
+
+        prev_epoch_block_hash = self.nodes[0].getblockhash(sc_creating_height - 1 + ((epoch_number_1) * EPOCH_LENGTH))
+
+        pkh_node1 = self.nodes[1].getnewaddress("", True)
+        self.sync_all()
+
+        epoch_block_hash_2, epoch_number_2 = get_epoch_data(scid_2, self.nodes[0], EPOCH_LENGTH)
+        amount_cert = [{"pubkeyhash": pkh_node1, "amount": bwt_amount}]
+
+        # Create Cert1 with quality 100 and place it in mempool
+        mark_logs("Create Cert1 with quality 100 and place it in mempool", self.nodes, DEBUG_MODE)
+        quality = 100
+        proof = mcTest.create_test_proof(
+            "sc1", epoch_number_1, epoch_block_hash_1, prev_epoch_block_hash,
+            quality, constant_1, [pkh_node1], [bwt_amount])
+        try:
+            cert_1_epoch_0 = self.nodes[0].send_certificate(scid_1, epoch_number_1, quality, epoch_block_hash_1, proof, amount_cert, CERT_FEE)
+            assert(len(cert_1_epoch_0) > 0)
+            mark_logs("Certificate is {}".format(cert_1_epoch_0), self.nodes, DEBUG_MODE)
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs("Send certificate failed with reason {}".format(errorString), self.nodes, DEBUG_MODE)
+            assert(False)
+
+        mark_logs("Checking mempools alignement", self.nodes, DEBUG_MODE)
+        self.sync_all()
+        for i in range(1, NUMB_OF_NODES):
+            assert_equal(sorted(self.nodes[0].getrawmempool()), sorted(self.nodes[i].getrawmempool()))
+
+        mark_logs("Check cert is in mempools", self.nodes, DEBUG_MODE)
+        assert_equal(True, cert_1_epoch_0 in self.nodes[0].getrawmempool())
+
+        # Create Cert2 with lower quality, any fee, deps on cert_1 and try to place it in mempool
+        mark_logs("#Checking rejection of cert_2 with same (scId, epoch), lower quality,  any fee, with spending deps on cert_1", self.nodes, DEBUG_MODE)
+        low_quality_proof = mcTest.create_test_proof(
+            "sc1", epoch_number_1, epoch_block_hash_1, prev_epoch_block_hash,
+            quality - 10, constant_1, [pkh_node1], [bwt_amount])
+        try:
+            cert_2_epoch_0 = self.nodes[0].send_certificate(scid_1, epoch_number_1, quality - 10, epoch_block_hash_1, low_quality_proof, amount_cert, CERT_FEE)
```

Comment:
When a dependancy between certs is required, it would be better using the rpc cmd 'createrawcertificate' which allows an explicit settings of the relevant input instead of relying on the src code implementation of the wallet, which implicitly uses previous cert change.